### PR TITLE
Return 500 if trying to run path with non-existing image

### PR DESCRIPTION
**Pull request by seiflotfy**
Fixes: #540

### CODE REVIEW

File: api/server/runner.go [hunk 1]
```diff
@@ -225,6 +225,12 @@ func (s *Server) serve(ctx context.Context, c *gin.Context, appName string, foun
 	default:
 		result, err := runner.RunTask(s.tasks, ctx, cfg)
 		if err != nil {
```
Comment (treeder) [inline api/server/runner.go]: We need to log this so the error isn't lost and someone can figure out what's wrong.

Comment (seiflotfy) [inline api/server/runner.go]: its already in the logs from the runner AFAICT
```ERRO[0003] Failed to pull image                          action="server.handleRunnerRequest)-fm" app=myapp call_id=357e93d0-af46-5cbb-83c6-7f8628fb2344 error="API error (404): {\"message\":\"repository iron/notexist not found: does not exist or no pull access\"}\n" image="iron/notexist" registry="https://registry.hub.docker.com" route="/lol2" username=```

**Comment by treeder** [comment]
Oh, ok. I'm thinking the runner probably shouldn't be logging that, it's up to the caller to decide what to do with the error. And what if at some point in the future, the runner decides to stop logging that?  Then it's lost in the void and turns into a debugging nightmare. cc @rdallman 

Recent Twitter thread I noticed on this: https://twitter.com/peterbourgon/status/834350104043741184

**Comment by rdallman** [comment]
@treeder we log and return the same error, wrt text we slightly improve the error before returning it so they're not the same value, but it's the same contextually. I don't really find twitter posts from some random guy as principles (yes i know who he is, we're all random guys). there are multiple cases in the runner where we want to log the error and it may also end up getting returned, due to the nature of retrying a lot of the docker calls, where we really do want to log it at the source. that being said, libraries that don't log stuff are ideal, but sometimes it's impractical. it would probably be nice to let callers specify some kind of logger output object in case they don't want logs, but at the same time this driver imo will never be used outside of our own devices. tl;dr doesn't really seem like a big deal ?

**Comment by treeder** [comment]
No, not a big deal at all, just saying we should log it too, just in case runner decides to not log it in the future and the error disappears into the void at that point.

File: api/server/runner.go (hunk 1 above)
Comment (seiflotfy) [inline api/server/runner.go]: Done!

**Pull request merged**